ci/scripts/build-arrow-cpp-minimal.sh

```diff
@@ -45,10 +45,10 @@ ARROW_CPP_SCRATCH_DIR="arrow-cpp-build-${ARROW_CPP_VERSION}"
 mkdir "${ARROW_CPP_SCRATCH_DIR}"
 pushd "${ARROW_CPP_SCRATCH_DIR}"
 
-curl -L "https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-${ARROW_CPP_VERSION}/apache-arrow-${ARROW_CPP_VERSION}.tar.gz" | \
+curl -L "https://github.com/apache/arrow/archive/refs/heads/main.tar.gz" | \
   tar -zxf -
 mkdir build && cd build
-cmake ../apache-arrow-${ARROW_CPP_VERSION}/cpp \
+cmake ../arrow-main/cpp \
```

Comment:
Don't think so. I fixed a few but they keep popping up, especially after having added metal / ipc support. Can keep trying but I think unless Arrow comprehensively wanted to fix these warnings its going to be hard downstream to keep this clean

An alternate solution may be to treat Arrow as a system include in the build systems so that it doesn't show any warnings from that library. That might allow us to at least keep this clean within nanoarrow